src/modules/75-cd/components/EnvironmentGroups/EnvironmentGroupsFilters/EnvironmentGroupsFilters.tsx

```diff
@@ -0,0 +1,248 @@
+        })
+      }
+      updateQueryParams({ page: [] as any, filters: JSON.stringify({ ...(filterFromFormData || {}) }) })
+      hideFilterDrawer()
+    } else {
+      showError(getString('filters.invalidCriteria'))
+    }
+  }
+
+  const handleSaveOrUpdate = async (
+    isUpdate: boolean,
+    data: FilterDataInterface<EnvironmentGroupFilterFormType, FilterInterface>
+  ) => {
+    setLoading(true)
+    const requestBodyPayload = createRequestBodyPayload({
+      isUpdate,
+      data,
+      projectIdentifier,
+      orgIdentifier
+    })
+
+    const saveOrUpdateHandler = filterRef.current?.saveOrUpdateFilterHandler
+    if (saveOrUpdateHandler && typeof saveOrUpdateHandler === 'function') {
+      const updatedFilter = await saveOrUpdateHandler(isUpdate, requestBodyPayload)
+      updateQueryParams({ filters: JSON.stringify({ ...(updatedFilter || {}) }) })
+    }
+
+    setLoading(false)
+    refetchFilters()
+  }
+
+  const handleDelete = async (filterIdentifier: string) => {
+    setLoading(true)
+    const deleteHandler = filterRef.current?.deleteFilterHandler
+    if (deleteHandler && typeof deleteFilter === 'function') {
+      await deleteHandler(filterIdentifier)
+    }
+    setLoading(false)
+
+    if (filterIdentifier === appliedFilter?.identifier) {
+      reset()
+    }
+    refetchFilters()
+  }
+
+  const handleFilterClick = (filterIdentifier: string) => {
+    if (filterIdentifier !== UNSAVED_FILTER_IDENTIFIER) {
+      updateQueryParams({
+        filterIdentifier,
+        filters: [] as any
+      })
+    }
+  }
+
+  const reset = () => {
+    replaceQueryParams({})
+  }
+
+  return (
+    <React.Fragment>
+      <FilterSelector<FilterDTO>
+        appliedFilter={appliedFilter}
+        filters={filters}
+        onFilterBtnClick={openFilterDrawer}
+        onFilterSelect={handleFilterSelection}
+        fieldToLabelMapping={fieldToLabelMapping}
+        filterWithValidFields={filterWithValidFields}
+      />
+      <Filter<EnvironmentGroupFilterFormType, FilterDTO>
+        isOpen={isFiltersDrawerOpen}
+        formFields={
+          <EnvironmentGroupsFilterForm environments={getMultiSelectFormOptions(environmentsResponse?.data?.content)} />
+        }
+        initialFilter={{
+          formValues: {
+            envGroupName,
+            description,
+            tags,
+            envIdentifiers
+          },
+          metadata: { name, filterVisibility, identifier, filterProperties: {} }
+        }}
+        filters={filters}
+        isRefreshingFilters={isFetchingFilters || isFetchingEnvironments || loading}
+        onApply={onApply}
+        onClose={() => hideFilterDrawer()}
+        onSaveOrUpdate={handleSaveOrUpdate}
+        onDelete={handleDelete}
+        onFilterSelect={handleFilterClick}
+        onClear={reset}
+        ref={filterRef}
+        dataSvcConfig={
+          new Map<CrudOperation, (...rest: any[]) => Promise<any>>([
```

Comment:
Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/harness/harness-core-ui/pullRequest?prid=9264786)
```suggestion
          new Map<CrudOperation, (...rest: unknown[]) => Promise<any>>([
```